File: server/models/medicine.js
const mongoose = require("mongoose");

const medicineSchema = mongoose.Schema({
    medicineName: {
        required: true,
        type: String,
        trim: true
    },
    salt: {
        required: true,
        type: String,
        trim: true
    },
    company: {
        required: false,
        type: String,
        trim: true
    },
    price: {
        required: true,
        type: Number,
        trim: true
    },
    quantity: {
        required: true,
        type: Number,
        trim: true
    },
    description: {
        required: false,
        type: String,
        trim: true
    }
});

const Medicine = mongoose.model("Medicine", medicineSchema);
module.exports = Medicine;

File: app/lib/models/medicine.dart
class Medicine {
  final String name;
  final String quantity;
  final String price;
  final String salt;
  final String company;
  final String description;
  final String id;

  Medicine(
      {required this.name,
      required this.quantity,
      required this.id,
      required this.price,
      required this.salt,
      required this.company,
      required this.description});

  static Medicine fromMap(Map<String, dynamic> data) {
    return Medicine(
        name: data['medicineName'],
        id:data['_id'],
        quantity: data['quantity'].toString(),
        salt: data['salt'],
        company: data['company'],
        description: data['description'],
        price: data['price'].toString());
  }
}
